org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/registry/ProjectRegistryManager.java

```diff
@@ -165,7 +156,7 @@
    */
   private final Map<MavenProjectFacade, MavenProject> legacyMavenProjects = new IdentityHashMap<>();
 
-  private final Cache<MavenProjectFacade, MavenProject> mavenProjectCache;
+  private final Map<MavenProjectFacade, MavenProject> mavenProjectCache;
```

Comment:
Should we use a ConcurrentMap here?

Reply:
I would like to use that too, but I only know the approach via the `LinkedHashMap` that is applied below to implement a capacity restricted Map.
If you know a way to achieve this with a `ConcurrentHashMap` in atomic operations please let me know.

Reply:
Then access should probably be synchronized, actually this shows that guava provides more with the cache as is provided "by standard java" and we either need to implement a cache our self or keep guava for this use-case.

Reply:
It is already synchronized, see the method `createProjectCache()`.
Most functionalities can be assembled with standard java, the only difference at the moment is that the concurrency-level is one while the Guava cache by default can handle up to 4 concurrent accesses, but this number is configurable.